opcua/common/node.py

```diff
@@ -617,14 +617,20 @@ def add_reference(self, target, reftype, forward=True, bidirectional=True):
         results =  self.server.add_references(params)
         _check_results(results, len(params))
 
-    def add_folder(self, nodeid, bname):
-        return  opcua.common.manage_nodes.create_folder(self, nodeid, bname)
+    def _add_modelling_rule(self, parent, mandatory):
+        if mandatory is not None and parent.get_node_class() == ua.NodeClass.ObjectType:
+            rule=ua.ObjectIds.ModellingRule_Mandatory if mandatory else ua.ObjectIds.ModellingRule_Optional
+            self.add_reference(rule, ua.ObjectIds.HasModellingRule, True, False)
+        return self
 
-    def add_object(self, nodeid, bname, objecttype=None):
-        return opcua.common.manage_nodes.create_object(self, nodeid, bname, objecttype)
+    def add_folder(self, nodeid, bname, mandatory=True):
+        return  opcua.common.manage_nodes.create_folder(self, nodeid, bname)._add_modelling_rule(self, mandatory)
 
-    def add_variable(self, nodeid, bname, val, varianttype=None, datatype=None):
-        return opcua.common.manage_nodes.create_variable(self, nodeid, bname, val, varianttype, datatype)
+    def add_object(self, nodeid, bname, objecttype=None, mandatory=True):
```

Comment:
Just a side comment. We need to be careful adding to many arguments to this methods. They were meant as a "dummy" way to add nodes. We should be carefull not to make them take 100 args.
What about making add_modelling_rule() public? instead of adding arguments?
But you can argument for adding this ' mandatory' arg.....

Reply:
Thiking about it. I think having modelling rule as optional is probably an exception. So I think we should really not add an argument but simply expose the add_modelling_rule() method

Reply:
The idea was to keep it backwards compatible.
I can refactor it to be stand-alone function, but then the user has to call it every time a node gets added to a type.

Another option is to just remove the argument and add the mandatory rule to all type children..
(Actually, this was my first implementation ;))

We could expose a `set_modelling_rule(bool)` function that is able to remove or change it afterwards.

Reply:
Exactly. Add automatically the mandatory rule. and let users that do not want it call set_modelling_rule. One might complain that this is not efficient, but those that want efficiency call use the opcua calls directly